components/script/dom/htmlimageelement.rs

```diff
@@ -782,6 +797,71 @@ impl LayoutHTMLImageElementHelpers for LayoutJS<HTMLImageElement> {
     }
 }
 
+pub fn parse_a_sizes_attribute(input: DOMString, width: Option<u32>) -> Result<Vec<Size>, String> {
+let mut sizes = Vec::<Size>::new();
+let unparsed_sizes = input.deref().split(',').collect::<Vec<_>>();
+
+    for unparsed_size in &unparsed_sizes {
+        let whitespace = unparsed_size.chars().rev().take_while(|c| char::is_whitespace(*c)).count();
+        let trimmed: String = unparsed_size.chars().take(unparsed_size.chars().count() - whitespace).collect();
+
+        if trimmed.is_empty() {
+            // return Err("reason".to_owned());
+            continue;
+        }
+        let mut input = ParserInput::new(&trimmed);
+        let url = ServoUrl::parse("about:blank").unwrap();
+        let reporter = NullReporter;
+        let context = ParserContext::new_for_cssom(&url,
+                                                        &reporter,
+                                                        None,
+                                                        ParsingMode::empty(),
+                                                        QuirksMode::NoQuirks);
+        let url = ServoUrl::parse("about:blank").unwrap();
```

Comment:
You already have `url` declared in line [813](https://github.com/Rakhisharma/servo/blob/bc95d0a8d75e5fbc943985e1e82e588c8d802d79/components/script/dom/htmlimageelement.rs#L813). So you can remove this.